### PR TITLE
feat: [Validation] add `field_exists` rule

### PR DESCRIPTION
**Pull request by kenjis**
~~Needs #8131~~

**Description**
For #8006

- add rule to check if the field exist in the data

**Checklist:**
- [x] Securely signed commits
- [x] Component(s) with PHPDoc blocks, only if necessary or adds value
- [ ] Unit testing, with >80% coverage
- [x] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

File: system/Validation/Rules.php [hunk 1]
```diff
@@ -353,4 +353,22 @@ public function required_without(
...
+     * @param string|null                             $field The field name.
+     */
+    public function field_exists(
+        $value = null,
+        ?string $param = null,
+        array $data = [],
+        ?string $error = null,
+        ?string $field = null
+    ): bool {
+        return array_key_exists($field, $data);
```
Comment (kenjis) [inline system/Validation/Rules.php:372]: Ah, we need to handle the key with dot array syntax...

Comment (MGatner) [reply system/Validation/Rules.php:372]: Good call, I'm afraid so. See `required_without` above. May be worth having a private method for `$data` transformations.

Comment (MGatner) [reply system/Validation/Rules.php:372]: Ah, I see your next PR - much better solution.

**Comment by michalsn** [comment]
I believe we would also need versions `_with` and `_without`.

Yes for dot array syntax.

File: system/Validation/Rules.php (hunk 1 above)
Comment (kenjis) [reply system/Validation/Rules.php:451]: Yes, done.

**Comment by kenjis** [comment]
> I believe we would also need versions `_with` and `_without`.

Really?
Do you have a use case?

**Comment by michalsn** [comment]
Sorry, I forgot about this one.

Example... an API with a custom payload? Some optional fields depend on selected parameters. They're not required though.

I think I saw a request like this a long time ago, but I can't find it now.

**Comment by kenjis** [comment]
I think we can use `required_with` in most cases.

> Some optional fields depend on selected parameters. They're not required though.

If it is not required, then it is okay that the key does not exists in the data.
Also, if it depends on selected parameters, we need to write custom rule?

If it is required, and the value may be null, empty string, then we need `field_exists_with`.

**Review by MGatner — APPROVED** [review]
Looks good to me. I think starting here makes sense and if we find need for additional related rules they can follow.

**Pull request merged**